pyproject.toml

```diff
@@ -1,7 +1,150 @@
 [build-system]
-requires = ["setuptools>=48", "setuptools_scm[toml]>=7"]
+requires = ["setuptools>=61", "setuptools_scm[toml]>=7"]
 build-backend = "setuptools.build_meta"
 
+[project]
+name = "dvc"
+description = "Git for data scientists - manage your code and data together"
+readme = "README.rst"
+keywords = [
+    "data-science",
+    "data-version-control",
+    "machine-learning",
+    "git",
+    "developer-tools",
+    "reproducibility",
+    "collaboration",
+]
+license = { file = "LICENSE" }
+authors = [{ name = "Dmitry Petrov", email = "dmitry@dvc.org" }]
+maintainers = [{ name = "Iterative", email = "support@dvc.org" }]
+requires-python = ">=3.8"
+dependencies = [
+    "colorama>=0.3.9",
+    "configobj>=5.0.6",
+    "voluptuous>=0.11.7",
+    "requests>=2.22.0",
+    # See https://github.com/bdcht/grandalf/issues/26
+    "grandalf==0.6",
+    "distro>=1.3.0",
+    "appdirs>=1.4.3",
+    "ruamel.yaml>=0.17.11",
+    "tomlkit>=0.11.1",
+    "funcy>=1.14",
+    "pathspec>=0.9.0,<0.10.0",
+    "shortuuid>=0.5.0",
+    "tqdm>=4.63.1,<5",
+    "packaging>=19.0",
+    "zc.lockfile>=1.2.1",
+    "flufl.lock>=5",
+    "networkx>=2.5",
+    "psutil>=5.8.0",
+    "pydot>=1.2.4",
+    "importlib-resources>=5.2.2; python_version < '3.9'",
+    "flatten_dict>=0.4.1,<1",
+    "tabulate>=0.8.7",
+    "pygtrie>=2.3.2",
+    "dpath>=2.0.2,<3",
+    "shtab>=1.3.4,<2",
+    "rich>=10.13.0",
+    "pyparsing>=2.4.7",
+    "typing-extensions>=3.7.4",
+    "scmrepo==0.1.2",
+    "dvc-render==0.0.12",
+    "dvc-task==0.1.4",
+    "dvclive>=0.10.0",
+    "dvc-data==0.25.2",
+    "dvc-http==2.27.2",
+    "hydra-core>=1.1.0",
+    "iterative-telemetry==0.0.5",
+]
+dynamic = ["version"]
+classifiers = [
+    "Development Status :: 4 - Beta",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+]
+
+[project.optional-dependencies]
+azure = ["dvc-azure==2.20.4"]
+gdrive = ["dvc-gdrive==2.19.0"]
+gs = ["dvc-gs==2.19.1"]
+hdfs = ["dvc-hdfs==2.19.0; python_version < '3.11'"]
+oss = ["dvc-oss==2.19.0"]
+s3 = ["dvc-s3==2.20.1"]
+ssh = ["dvc-ssh==2.20.0"]
+ssh_gssapi = ["dvc-ssh[gssapi]==2.20.0"]
+webdav = ["dvc-webdav==2.19.0"]
+webhdfs = ["dvc-webhdfs==2.19.0"]
+webhdfs_kerberos = ["dvc-webhdfs[kerberos]==2.19.0"]
+terraform = ["tpi[ssh]>=2.1.0"]
+testing = ["pytest-test-utils==0.0.8"]
+all = [
```

Comment:
This can be just `dvc[azure,gdrive,gs,s3,webdav,webhdfs,ssh,hdfs,oss]` if we require `pip>=21.2`.

Reply:
We will quickly forget to update this. Let's just switch to the new syntax. Esp if we can write down that we require `pip>=21.2`